parcels/field.py

```diff
@@ -131,39 +131,44 @@ class Field(object):
 
     :param name: Name of the field
     :param data: 2D, 3D or 4D array of field data
-    :param lon: Longitude coordinates of the field
-    :param lat: Latitude coordinates of the field
-    :param depth: Depth coordinates of the field
-    :param time: Time coordinates of the field
+    :param grid: :class:`parcels.grid.Grid` object containing all the lon, lat depth, time
+           mesh and time_origin information
+    :param lon: Longitude coordinates of the field. (only if grid is None)
+    :param lat: Latitude coordinates of the field. (only if grid is None)
+    :param depth: Depth coordinates of the field. (only if grid is None)
+    :param time: Time coordinates of the field. (only if grid is None)
     :param transpose: Transpose data to required (lon, lat) layout
     :param vmin: Minimum allowed value on the field.
            Data below this value are set to zero
     :param vmax: Maximum allowed value on the field
            Data above this value are set to zero
-    :param time_origin: Time origin of the time axis
+    :param time_origin: Time origin of the time axis (only if grid is None)
     :param units: type of units of the field (meters or degrees)
     :param interp_method: Method for interpolation
     :param allow_time_extrapolation: boolean whether to allow for extrapolation
     :param time_periodic: boolean whether to loop periodically over the time component of the Field
            This flag overrides the allow_time_interpolation and sets it to False
     """
 
-    def __init__(self, name, data, lon=None, lat=None, grid=None, depth=None, time=None,
+    def __init__(self, name, data, grid=None, lon=None, lat=None, depth=None, time=None,
```

Comment:
I didn't realise that swapping the order here actually breaks backward compatibility for codes that call `Field` still with `lon`, `lat` etc. If that's the case, then better to revert to what you had before? Or even put the `grid` argument after the time argument?

Reply:
Ok, I changed it back.
But only here (I didn't reverted the tests. Indeed, lon and lat are now optional so it is a bad practice not to put explicitly lon=lon, even it it works for backward compatibility)